### ISSUE TITLE
Tenant hostname is not checked for existence if the other tenant has multiple hostnames

### ISSUE THREAD

**Issue by Piedone** [OrchardCMS/OrchardCore#10269]
### To Reproduce

1. Create a tenant with the hostnames "example1.com, example2.com".
2. Create a tenant with the hostname "example1.com".
3. Observe that you can create the second tenant even though the hostname is duplicated.

### Expected behavior
I get the "A tenant with the same host and prefix already exists." validation error and I can't save the tenant.

Note that the validation error is raised if the first tenant has just a single hostname specified.

**Comment by Piedone** [comment]
FYI @agriffard created this.

**Comment by hishamco** [comment]
> Note that the validation error is raised if the first tenant has just a single hostname specified.

@Piedone it is a matter if the first tenant has single hostname or not? becuse the hostname can't be same for two tenant, right?

**Comment by Piedone** [comment]
It is not a matter then. I.e. this is correct:

1. Create a tenant with the hostnames "example.com" (i.e. a single host).
2. Create another tenant with the hostname "example.com".
3. Observe that you get the "A tenant with the same host and prefix already exists." validation error.

**Comment by hishamco** [comment]
Seems there are some cases are valid even the hostanme is duplicate. The following table shows some scenarios where the duplicate is valid or not:

|     Name |     Url Prefix |     Host Name |     Is Valid |
|----------- |----------:|----------:|------:|
| Tenant1 | tenant1 | example1.com | True |
| Tenant2 |  | example1.com,example2.com | True |
| Tenant3 |  tenant1 | example1.com | False |
| Tenant4 |   | example1.com | False |
| Tenant5 |   | example2.com | False |
| Tenant6 |  tenant1 | example2.com | True |
| Tenant7 |  tenant2 | example2.com | True |

**Comment by hishamco** [comment]
BTW the above scenario is already fixed, I check it and no problem is there, except there case are not covered. For instance when the host contains multiple domains

**Comment by Piedone** [comment]
Yeah, that's what the issue is about ;). Simply duplicate host names are not a problem, of course, but the host name + prefix combo should be unique, including multiple host names.

BTW having example.com/tenant1 for one tenant and example.com without a prefix for another tenant is valid and currently correctly allowed.
